### PR TITLE
[UI] save file change only on ctrl-s

### PR DESCRIPTION
**Pull request by mykalmax**

### CODE REVIEW

File: web/client/src/library/components/editor/CodeEditor.tsx [hunk 1]
```diff
@@ -24,37 +24,47 @@ export default function CodeEditor({
...
+      HoverTooltip(models),
+      events(models, files, selectFile),
+      SqlMeshModel(models),
       showSqlMeshDialect && SqlMeshDialect(models, file, sqlDialectOptions),
       file.extension === '.py' && python(),
       file.extension === '.yaml' && StreamLanguage.define(yaml),
+      keymap.of([{
+        mac: 'Cmd-s',
+        win: 'Ctrl-s',
+        linux: 'Cmd-s',
```
Comment (vchan) [inline web/client/src/library/components/editor/CodeEditor.tsx:58]: Isn't the command key mac only?

File: web/client/src/library/components/editor/Editor.tsx [hunk 2]
```diff
@@ -134,36 +123,33 @@ export default function Editor({
...
     onSuccess(file: File) {
       setIsSaved(true)
-      setEditorFileStatus(EnumEditorFileStatus.Edit)
 
       if (file == null) return
 
-      activeFile.content = file.content ?? ''
-
+      activeFile.updateContent(file.content)
+
```
Comment (vchan) [inline web/client/src/library/components/editor/Editor.tsx:137]: Does prettier remove these whitespaces?

File: web/client/src/models/file.ts [hunk 3]
```diff
@@ -53,4 +56,9 @@ export class ModelFile extends ModelArtifact<InitialFile> {
   get isSQLMeshSeed(): boolean {
     return ['.csv'].includes(this.extension) && this.path.startsWith('seeds/')
   }
+
+  updateContent(newContent: string = ''): void {
```
Comment (vchan) [inline web/client/src/models/file.ts:60]: Is content updated when user types into the editor?

**Review by vchan — APPROVED** [review]

File: web/client/src/library/components/editor/Editor.tsx (hunk 2 above)
Comment (mykalmax) [reply web/client/src/library/components/editor/Editor.tsx:137]: it was disabled

File: web/client/src/models/file.ts (hunk 3 above)
Comment (mykalmax) [reply web/client/src/models/file.ts:60]: yes, but we also need to reset `_content` when we saving file content to keep track of changed state

**Pull request merged**